pinot-core/src/main/java/org/apache/pinot/core/common/BlockDocIdSet.java

```diff
@@ -21,6 +21,4 @@
 public interface BlockDocIdSet {
 
   BlockDocIdIterator iterator();
```

Comment:
Java doc over this method and/or the interface? Since we are refactoring this interface, it is a good time to add javadoc for a public interface as well.

Reply:
Good point, will add javadoc

Reply:
can you also explain why it is called block*?

Reply:
The result from the Operator is called Block, and BlockDocIdSet is the document ids from a Block. It is not structured very well, but it is out of the scope of this PR. We have plan to clean that up later.